salt/modules/zypper.py

```diff
@@ -1702,8 +1796,9 @@ def list_locks():
         salt '*' pkg.list_locks
     '''
     locks = {}
-    if os.path.exists(LOCKS):
-        with salt.utils.files.fopen(LOCKS) as fhr:
+    _locks = os.path.join(root, os.path.relpath(LOCKS, start='/')) if root else LOCKS
+    if os.path.exists(_locks):
+        with salt.utils.files.fopen(_locks) as fhr:
```

Comment:
I'd rather try/except it instead of stat file twice.

Reply:
done, complicate the code a bit